src/analysis/processing/qgsalgorithmalignrasters.cpp

```diff
@@ -0,0 +1,182 @@
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgsalgorithmalignrasters.h"
+#include "qgsprocessingparameteralignrasterlayers.h"
+#include "qgsalignraster.h"
+#include "qgsalignrasterdata.h"
+
+///@cond PRIVATE
+
+QString QgsAlignRastersAlgorithm::name() const
+{
+  return QStringLiteral( "alignrasters" );
+}
+
+QString QgsAlignRastersAlgorithm::displayName() const
+{
+  return QObject::tr( "Align rasters" );
+}
+
+QStringList QgsAlignRastersAlgorithm::tags() const
+{
+  return QObject::tr( "raster,align,resample,rescale" ).split( ',' );
+}
+
+QString QgsAlignRastersAlgorithm::group() const
+{
+  return QObject::tr( "Raster tools" );
+}
+
+QString QgsAlignRastersAlgorithm::groupId() const
+{
+  return QStringLiteral( "rastertools" );
+}
+
+QString QgsAlignRastersAlgorithm::shortHelpString() const
+{
+  return QObject::tr( "Aligns rasters by resampling them to the same cell size and reprojecting to the same CRS." );
+}
+
+QgsAlignRastersAlgorithm *QgsAlignRastersAlgorithm::createInstance() const
+{
+  return new QgsAlignRastersAlgorithm();
+}
+
+void QgsAlignRastersAlgorithm::initAlgorithm( const QVariantMap & )
+{
+  addParameter( new QgsProcessingParameterAlignRasterLayers( QStringLiteral( "LAYERS" ), QObject::tr( "Input layers" ) ) );
+  addParameter( new QgsProcessingParameterRasterLayer( QStringLiteral( "REFERENCE_LAYER" ), QStringLiteral( "Reference layer" ) ) );
+
+  addParameter( new QgsProcessingParameterCrs( QStringLiteral( "CRS" ), QObject::tr( "CRS" ), QVariant(), true ) );
+  addParameter( new QgsProcessingParameterNumber( QStringLiteral( "CELL_SIZE_X" ), QObject::tr( "Cell size X" ), QgsProcessingParameterNumber::Double, QVariant(), true, 1e-9 ) );
+  addParameter( new QgsProcessingParameterNumber( QStringLiteral( "CELL_SIZE_Y" ), QObject::tr( "Cell size Y" ), QgsProcessingParameterNumber::Double, QVariant(), true, 1e-9 ) );
+  addParameter( new QgsProcessingParameterNumber( QStringLiteral( "GRID_OFFSET_X" ), QObject::tr( "Grid offset X" ), QgsProcessingParameterNumber::Double, QVariant(), true, 1e-9 ) );
+  addParameter( new QgsProcessingParameterNumber( QStringLiteral( "GRID_OFFSET_Y" ), QObject::tr( "Grid offset Y" ), QgsProcessingParameterNumber::Double, QVariant(), true, 1e-9 ) );
+  addParameter( new QgsProcessingParameterExtent( QStringLiteral( "EXTENT" ), QObject::tr( "Clip to extent" ), QVariant(), true ) );
```

Comment:
I think the descriptions of these could hint that they override the reference layer properties. 

Ie instead of "CRS", something like "Override reference CRS" ?